### PR TITLE
Add Course Layer

### PR DESCRIPTION
**Pull request by marcgabe15**
# Description

This PR adds in the courses layer with:
- courses routes
- SQL connection and new database connection
- A definition for the course model
- course controllers and a data access layer

Moving forward, the way implement a route and access the data layer is:
- create a file for your model and define attributes under ```model/```
- under ```dal/``` create a function to do CRUD operations against model
- create a controller file to handle request and response
- route via router under v2
- 
# Motivation and Context
- this adds in the courses layer for doing multi class integration

# How Has This Been Tested?
- manual for now
- TODO: create unit tests

# Screenshots (if appropriate):

### CODE REVIEW

File: src/api/controllers/course.controller.ts [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+};
+
+export const getCourseById = async (
+  req: Request,
+  res: Response
+): Promise<void> => {
+  try {
+    const id = req.params['courseId'];
+    if (id === undefined) {
+      res.status(500).send('error course is undefined');
```
Comment (kapooramanpreet) [inline src/api/controllers/course.controller.ts:62]: I think this should be 400 error. same for update and delete (line 23, 43)

**Review by kapooramanpreet — COMMENTED** [review]
Rest looks good.

File: src/api/models/course.model.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
+import { DataTypes, Model, Optional } from 'sequelize';
+import { sequelize } from '../../config/databasev2';
+import { Organization } from './organization.model';
+export interface CourseAttributes {
+  id: string;
+  courseName: string;
```
[the remark this answers is not in the document]
Comment (prayujt) [reply src/api/models/course.model.ts:6]: I will be adding the LTI info for problems and courses in my LTI PR

File: src/api/controllers/course.controller.ts (hunk 1 above)
Comment (prayujt) [reply src/api/controllers/course.controller.ts:62]: fixed in #145

File: src/config/databasev2.ts [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import { Sequelize } from 'sequelize';
```
[the remark this answers is not in the document]
Comment (prayujt) [reply src/config/databasev2.ts:1]: fixed in #145

**Comment by marcgabe15** [comment]
closing and using Prayuj's pull request instead

**Pull request closed without merging**